cartographer/transform/transform_interpolation_buffer.h

```diff
@@ -27,18 +28,29 @@
 namespace cartographer {
 namespace transform {
 
+constexpr size_t kUnlimitedBufferSize = std::numeric_limits<size_t>::max();
+
 // A time-ordered buffer of transforms that supports interpolated lookups.
+// Unless explicitly set, the buffer size is unlimited.
 class TransformInterpolationBuffer {
  public:
   TransformInterpolationBuffer() = default;
   explicit TransformInterpolationBuffer(
       const mapping::proto::Trajectory& trajectory);
+  explicit TransformInterpolationBuffer(const size_t buffer_size_limit);
```

Comment:
Not sure this constructor provides much value. Convenience is not necessarily bad, but here it seems to also harm readability as you would call the constructor with just an integer. Calling `SetSizeLimit()` directly after construction makes it much clearer what is happening. Remove?

Reply:
Sounds good.